lxd/storage/drivers/driver_zfs.go

```diff
@@ -356,12 +356,6 @@ func (d *zfs) Create() error {
 
 	revert.Add(func() { _ = d.Delete(nil) })
 
-	// Apply our default configuration.
-	err = d.ensureInitialDatasets(false)
```

Comment:
We need to keep this so that if the initial datasets cannot be created and configured then the pool creation will fail.

Mount() allows for warning if something goes wrong to allow for historically setup pools to not start failing if new requirements change in the future.